player/lua/console.lua

```diff
@@ -239,6 +239,21 @@ function ass_escape(str)
     return mp.command_native({'escape-ass', str})
 end
 
+local function get_scaled_osd_dimensions()
+    local w, h, aspect = mp.get_osd_size()
+
+    if mp.get_property_native('osd-scale-by-window') then
+        h = 720
+        w = 720 * aspect
+    end
+
+    local scale = mp.get_property_native("display-hidpi-scale", 1) * opts.scale
```

Comment:
Do we remove it then?

Reply:
I also removed `scale` since I don't understand its purpose.

Reply:
`scale` is basically required for readability on a hidpi display for lower-resolution videos. Well for me at least.

Reply:
How does it differ from increasing `font_size`?

Reply:
It works out of the box and follows your OS scale settings.

Reply:
Are you talking about `scale` or `display-hidpi-scale`?

Reply:
>Are you talking about scale or display-hidpi-scale?

The latter to be exact. I'm not sure if the script option is useful, but automatically adjusting font size based on your OS's DPI is.

>scale doesn't really solve the problem.

I'm not sure what you mean but with `hidpi-window-scale` it is perfectly readable to me. This PR is a regression imo for hidpi displays.

Reply:
I don't mind either way if we keep using `display-hidpi-scale` but then why don't the OSD and other scripts use it?

Reply:
No one ever implemented it that way. ¯\_(ツ)_/¯ AFAIK, console always had it.

Reply:
Readded `display-hidpi-scale` as I think it makes more sense to factor it in all text sizes instead or removing it from console.lua.

Reply:
What's the conclusion? Everyone is happy with current patch? +1 from me.

Reply:
No, I'm just asking, if it needs to wait, there is not rush.

Reply:
I'm not really sure scaling the console with the window height is desirable tbh. It's just text. Does your browser's text size change when you change the window size?

Reply:
I don't get why we have to be the only application to scale text with window height either and use `--osd-scale-by-window=no`. but both kasper and nanahi requested this.

Reply:
You guys are moving the bar right now. Everything in mpv scales. OSD, OSC, stats, subtitles. Name one thing which does not scale... console. I requested this, because it makes sense it mpv context.

Reply:
tbh there could be an argument made for subtitles not scaling. But anyways, I don't see the advantage of this change. It's fine to have it as an option I guess, but to me it's just a strictly worse default imo

Reply:
I think this scaling only makes sense to not break ASS subtitles and they are not affected by `--sub-scale-by-window` anyway (despite the docs stating "Like --sub-scale, this can break ASS subtitles."). I don't understand the advantage of scaling by default elsewhere.